drivers/cc2420/cc2420.c

```diff
@@ -0,0 +1,231 @@
+
+    /* Power on and toggle reset */
+    gpio_set(dev->power_pin);
+    gpio_clear(dev->reset_pin);
+    xtimer_usleep(CC2420_WAIT_TIME);
+    gpio_set(dev->reset_pin);
+
+    /* enable transceiver's crystal oscillator */
+    cc2420_strobe(dev, CC2420_STROBE_XOSCON);
+
+    /* wait for the oscillator to be stable */
+    xtimer_usleep(CC2420_WAIT_TIME);
+    if ((CC2420_STATUS_BYTE() & CC2420_STATUS_XOSC16M_STABLE) == 0) {
+        /* could not start up radio transceiver! */
+        core_panic(0x2420, "Could not start CC2420 radio transceiver");
+    }
+
+    /* discard any potential garbage in TX buffer */
+    cc2420_strobe(dev, CC2420_STROBE_FLUSHTX);
+
+    /* reset options and sequence number */
+    dev->seq_nr = 0;
+    dev->options = 0;
+    /* set short and long address */
+#if CPUID_ID_LEN
+    cpuid_get(cpuid);
+
+#if CPUID_ID_LEN < 8
+    /* in case CPUID_ID_LEN < 8, fill missing bytes with zeros */
+    for (int i = CPUID_ID_LEN; i < 8; i++) {
+        cpuid[i] = 0;
+    }
+#else
+    for (int i = 8; i < CPUID_ID_LEN; i++) {
+        cpuid[i & 0x07] ^= cpuid[i];
+    }
+#endif
+    /* make sure we mark the address as non-multicast and not globally unique */
+    cpuid[0] &= ~(0x01);
+    cpuid[0] |= 0x02;
+    /* copy and set long address */
+    memcpy(&addr_long, cpuid, 8);
+    cc2420_set_addr_long(dev, NTOHLL(addr_long.uint64.u64));
+    cc2420_set_addr_short(dev, NTOHS(addr_long.uint16[0].u16));
+#else
+    cc2420_set_addr_long(dev, CC2420_DEFAULT_ADDR_LONG);
+    cc2420_set_addr_short(dev, CC2420_DEFAULT_ADDR_SHORT);
+#endif
+    /* set default PAN id */
+    cc2420_set_pan(dev, CC2420_DEFAULT_PANID);
+    /* set default channel */
+    cc2420_set_chan(dev, CC2420_DEFAULT_CHANNEL);
+    /* set default TX power */
+    cc2420_set_txpower(dev, CC2420_DEFAULT_TXPOWER);
+    /* set default options */
+    cc2420_set_option(dev, CC2420_OPT_AUTOACK, true);
+    cc2420_set_option(dev, CC2420_OPT_CSMA, true);
+    cc2420_set_option(dev, CC2420_OPT_TELL_RX_START, false);
+    cc2420_set_option(dev, CC2420_OPT_TELL_RX_END, true);
+    /* set default protocol */
+#ifdef MODULE_NG_SIXLOWPAN
+    dev->proto = GNRC_NETTYPE_SIXLOWPAN;
+#else
+    dev->proto = GNRC_NETTYPE_UNDEF;
+#endif
+
+    /* Change default values as recomended in the data sheet, */
+    /* RX bandpass filter = 1.3uA. */
+    uint16_t reg = cc2420_reg_read(dev, CC2420_REG_RXCTRL1);
+    reg |= CC2420_RXBPF_LOCUR;
+    cc2420_reg_write(dev, CC2420_REG_RXCTRL1, reg);
+
+    /* Set the FIFOP threshold to maximum. */
+    cc2420_reg_write(dev, CC2420_REG_IOCFG0, 127);
+
+    /* Turn off "Security enable" (page 32). */
+    reg = cc2420_reg_read(dev, CC2420_REG_SECCTRL0);
+    reg &= ~CC2420_RXFIFO_PROTECTION;
+    cc2420_reg_write(dev, CC2420_REG_SECCTRL0, reg);
+
+    /* go into RX state */
+    cc2420_set_state(dev, CC2420_STATE_RX_ON);
+}
+
+bool cc2420_cca(cc2420_t *dev)
+{
+    long count = 0;
+    do {
+        count++;
+        if (count >= CC2420_WAIT_TIME) {
```

Comment:
this use of wait time seems off, you use `CC2420_WAIT_TIME` as argument for xtimer earlier, and now as counter max value?